qctrlopencontrols/dynamic_decoupling_sequences/predefined.py

```diff
@@ -59,6 +69,25 @@ def _add_pre_post_rotations(
         resulting after the addition of pi/2 pulses at the start and end of the sequence.
     """
 
+    # Setting the azimuthal angle of the final pi/2-pulse as 0 by default,
+    # which is the same value as the initial pi/2-pulse
+    final_azimuthal = 0.
+
+    # These lists have 1 if the pulse is of the specificed type, 0 if it is not
+    is_x_pi_pulse = np.where(np.logical_and(np.isclose(rabi_rotations, np.pi),
```

Comment:
optional: instead of effectively `sum(np.where(..., 1, 0))`, you could do `np.count_nonzero(...)`.

Reply:
That's interesting, I didn't know `True`/`False` would count as 1/0 over there. Thanks for the tip!